plugin.video.jafc/resources/lib/search.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+
+"""JAFC searcher and helpers"""
+__author__ = "fraser"
+
+import json
+
+import requests
+from bs4 import BeautifulSoup
+
+from . import kodiutils as ku
+from .cache import Cache, datetime_to_httpdate
+
+JAFC_URI = "https://animation.filmarchives.jp/"
+JAFC_SEARCH_URI = JAFC_URI + "en/works/"
+JAFC_INFO_URI = JAFC_URI + "/en/works/view/"
+JAFC_M3U8_TEMPLATE = "https://h10.cs.nii.ac.jp/stream/nfc/{}_en/hls/auto/media-2/stream.m3u8"
+
+
+SEARCH_SAVED = ku.get_setting_as_bool("search_saved")
+SEARCH_MAX_RESULTS = ku.get_setting_as_int("search_max_results")
+SEARCH_TIMEOUT = 60
+
+CACHE_URI = ku.translate_path("special://profile/addon_data/plugin.video.jafc/cache.sqlite")
```

Comment:
Please use API, 
```
import xbmcaddon
ADDON = xbmcaddon.Addon()
ADDON_DATA = ADDON .getAddonInfo('profile').decode('utf-8')
CACHE_URI = os.path.join(ADDON_DATA,'cache.sqlite')
```

Reply:
This seems arbitrary rather than objective - the other three plugins I have submitted that have been accepted use *exactly* the same code. Also - why on earth would I use os.path to join a SPECIAL url? That doesn't seem to make any sense.

Reply:
When I try your suggested code I get.

File "/Users/fraser/Library/Application Support/Kodi/addons/plugin.video.jafc/resources/lib/cache.py", line 193, in _execute
                                                with self.connection:
                                            AttributeError: __exit__

When I do 

CACHE_URI = ku.translate_path(os.path.join(ADDON_DATA,'cache.sqlite'))

Then it works...really not sure how to implement your suggested change as it breaks the code :(

Reply:
FWIW 

os.path.join(ADDON_DATA,'cache.sqlite')  is equal to "special://profile/addon_data/plugin.video.jafc/cache.sqlite"

Reply:
You don't need to execute translatepath on the paths provided by `xbmcaddon.Addon().getAddonInfo('profile')` or `xbmcaddon.Addon().getAddonInfo('path')`, they will return an already translated path. You only need to translate special:// paths. Your profile special path will be equal (after translation) to the one obtained from `getAddonInfo`. If they were not the same something was tremendously wrong :)

Reply:
@enen92 - Sure that makes sense :) I think the confusion arose from the fact that the example given in the initial review was wrong.